examples/SimpleRagas/SimpleRagas.ipynb

```diff
@@ -0,0 +1,1018 @@
+    "        score = 0.0\n",
+    "\n",
+    "    braintrust.current_span().log(\n",
+    "        input={\"query\": query, \"document\": document},\n",
+    "        output=score,\n",
+    "    )\n",
+    "\n",
+    "    return score\n",
+    "\n",
+    "\n",
+    "async def retrieval_qa_w_context_and_llm_context_selection(input):\n",
+    "    embedding = await embed_text(input)\n",
+    "\n",
+    "    with braintrust.current_span().start_span(\n",
+    "        name=\"vector search\", input=input\n",
+    "    ) as span:\n",
+    "        result = table.search(embedding).limit(TOP_K_TO_FETCH).to_arrow().to_pylist()\n",
+    "        docs = [markdown_sections[i[\"section_id\"]][\"markdown\"] for i in result]\n",
+    "\n",
+    "        relevance_scores = []\n",
+    "        for doc in docs:\n",
+    "            relevance_scores.append(await relevance_score(input, doc))\n",
+    "\n",
+    "        span.log(\n",
+    "            output=[\n",
+    "                {\n",
+    "                    \"doc\": markdown_sections[r[\"section_id\"]][\"markdown\"],\n",
+    "                    \"distance\": r[\"_distance\"],\n",
+    "                }\n",
+    "                for r in result\n",
+    "            ],\n",
+    "            metadata={\"top_k\": TOP_K_TO_FETCH, \"retrieval\": result},\n",
+    "            scores={\n",
+    "                \"avg_relevance\": sum(relevance_scores) / len(relevance_scores),\n",
+    "                \"min_relevance\": min(relevance_scores),\n",
+    "                \"max_relevance\": max(relevance_scores),\n",
+    "            },\n",
+    "        )\n",
+    "\n",
+    "    # return top 3 docs by relevance_score:\n",
+    "    top_docs = [\n",
+    "        docs[i]\n",
+    "        for i in sorted(\n",
+    "            range(len(relevance_scores)), key=lambda i: -relevance_scores[i]\n",
+    "        )[:TOP_K_TO_USE]\n",
+    "    ]\n",
+    "\n",
+    "    context = \"\\n------\\n\".join(top_docs)\n",
+    "    completion = await client.chat.completions.create(\n",
+    "        model=QA_ANSWER_MODEL,\n",
+    "        messages=[\n",
+    "            {\n",
+    "                \"role\": \"user\",\n",
+    "                \"content\": f\"\"\"\\\n",
+    "                Given the following context\n",
+    "                \n",
+    "                {context}\n",
+    "                \n",
+    "                Please answer the following question:\n",
+    "                \n",
+    "                Question: {input}\n",
+    "                \"\"\",\n",
+    "            }\n",
+    "        ],\n",
+    "    )\n",
+    "    answer = completion.choices[0].message.content\n",
+    "\n",
+    "    return dict(context=context, n_contexts=len(top_docs), answer=answer)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now run our end-to-end pipeline and store the question, our generated answer, context used, and the ground truth answer:\n"
```

Comment:
A better way to do this is, in the task function, log metadata using the `hooks` argument that's passed in. That way you don't need to have a pre-processing step that builds up the context in the dataset, but you'll still be able to use it in the scoring functions.